src/server/index.js

```diff
@@ -90,15 +90,29 @@ app.get('/api/vertical', (req, res) => {
 })
 
 app.post('/api/heatmap', (req, res) => {
-  adapter
-    .heatMap(req.body.genes)
-    .then(function (results) {
-      res.json(results.rows)
-    })
-    .catch(function (err) {
-      console.log(err)
-      res.status(500).json({ status: 'Error' })
-    })
+  if (req.body.type === 'fpkm') {
+    adapter
```

Comment:
Consider something like:

```
const heatmap = type === 'fpkm' ? adapter.heatMap(...) : adapter.intronAnalysisHeatmap(...)
heatmap
.then(function (...
```

This way you don't repeat the `then` and `catch` code.

Reply:
Feel free to use `if...else` if you don't like the ternary operator.